lazy.js

```diff
@@ -6628,6 +6628,37 @@
     return snapshot;
   };
 
+  Sequence.prototype.flatten = function flatten () {
+    var source
+    var queue
+    var results
+    var item
+
+    source = this.source
+    queue = []
+    results = []
+
+    source.forEach(it => queue.push(it))
+
+    while (queue.length) handle(queue.shift())
+
+    this.source = results
```

Comment:
So this mutates the current sequence. This isn't how this library works. Take a look at any of the other methods for a better sense of how to implement this.